src/lib/services/feature-toggle-service.ts

```diff
@@ -628,35 +647,35 @@ class FeatureToggleServiceV2 {
     async updateStale(
         featureName: string,
         isStale: boolean,
-        userName: string,
+        createdBy: string,
     ): Promise<any> {
         const feature = await this.featureToggleStore.get(featureName);
         feature.stale = isStale;
         await this.featureToggleStore.update(feature.project, feature);
         const tags = await this.featureTagStore.getAllTagsForFeature(
             featureName,
         );
-        const data = await this.getFeatureToggleLegacy(featureName);
 
         await this.eventStore.store({
```

Comment:
I am considering creating types for all events, just to make it easier to remember to add all the the correct metadata fields. E.g. 

```js
interface FeatureStaleOnEvent {
 type: 'feature-stale-on';
 featureName: string;
 project: string;
 tags:  ITag[];
 createdBy: name;
}
```

Reply:
I created an example of this in: https://github.com/Unleash/unleash/pull/1089/commits/2f81271898bd2f4665a6358768f5309092170ab4

Hope to get some feedback on the implementation before rolling it for all events. It adds type safety and makes it "impossible" to provide required data for different event types.

Reply:
Nice. It will also make it easier to refactor events later. I like it.